src/buildbot.py

```diff
@@ -38,7 +39,7 @@ def Fail(warn_only=False):
   """Mark one step as failing, but keep going."""
   sys.stdout.flush()
   if warn_only:
-    sys.stdout.write('\n@@@STEP_WARNINGS@@@\n')
+    Warn()
```

Comment:
1. I think it would be more clear if we kept Fail() just for hard failures, and then have Warn() do what Fail(warn_only=True) does today.
--> I can do that

2. The other option might be to keep the boolean flags, and put the printing together in one function with a boolean switch for the type.
--> The reason I made Warn() is I wanted buildbot.WarnedList() (to be used for condition whether to test JSC tests), and it would look weird if they don't have symmetric methods for Fail() and Failed(). If we delete Warn() and move the printing to Fail(), the symmetry would break, but I don't mind too much either.